src/prefect/task_runners.py

```diff
@@ -282,3 +286,43 @@ def __exit__(self, exc_type, exc_value, traceback):
             self._executor.shutdown()
             self._executor = None
         super().__exit__(exc_type, exc_value, traceback)
+
+
+class PrefectTaskRunner(TaskRunner[PrefectDistributedFuture]):
+    def duplicate(self) -> "PrefectTaskRunner":
+        return type(self)()
+
+    def submit(
+        self,
+        task: "Task",
+        parameters: Dict[str, Any],
+        wait_for: Optional[Iterable[PrefectFuture]] = None,
+        dependencies: Optional[Dict[str, Set[TaskRunInput]]] = None,
+    ) -> PrefectDistributedFuture:
+        """
+        Submit a task to the Prefect API for a task server to run.
+
+        Args:
+            task: The task to submit.
+            parameters: The parameters to use when running the task.
+            wait_for: A list of futures that the task depends on.
+
+        Returns:
+            A future object that can be used to wait for the task to complete and
+            retrieve the result.
+        """
+        if not self._started:
+            raise RuntimeError("Task runner is not started")
+
+        from prefect.context import FlowRunContext
+
+        flow_run_ctx = FlowRunContext.get()
+        if flow_run_ctx:
+            get_run_logger(flow_run_ctx).info(
+                f"Submitting task {task.name} to Prefect API..."
+            )
+        else:
+            self.logger.info(f"Submitting task {task.name} to Prefect API...")
+
+        # TODO: wait_for?
+        return task.apply_async(kwargs=parameters)
```

Comment:
We may have to implement `wait_for` in `apply_async` in order to satisfy the same task runner interface for `submit`